_posts/2021-11-23-smallrye-stork-intro.adoc

```diff
@@ -0,0 +1,68 @@
+---
+layout: post
+title: 'SmallRye Stork meets Quarkus'
+date: 2021-11-23
+tags: extension smallrye-stork
+synopsis: Introduction to customizable service discovery and client side load balancing with Quarkus and SmallRye Stork
+author: michalszynkiewicz
+---
+:imagesdir: /assets/images/posts/smallrye-stork
+
+Last week, the Quarkus Insights spotlighted a new project called SmallRye Stork and its integration in Quarkus. In this blog post, we're going to give you a concise introduction to Stork.
+
+== What's the problem?
+
+Nowadays, systems we build consist of numerous applications, often called microservices. The system's overall behavior emerges from the interactions between all these services. So, a microservice is likely going to invoke other ones. But here is the catch. In dynamic environments, such as Kubernetes, or the Cloud, the locations of these services may change over time. So, how does our microservice locate the services it needs to call?
+
+Service discovery solves that problem, as it provides a mechanism to register and find the services. But it leads to another issue. What happens when there are multiple instances of the same service? How does a microservice select the right one? This scenario typically occurs when there are multiple replicas or versions available at the same time. Thus, in addition to service discovery, we need a way to select the right one.
+
+Some environments, e.g., Kubernetes, provide ways to discover service locations and load-balance the traffic. Some do not. In some cases, the capabilities of the environment are not sufficient for the needs of the system.
+
+image::stork.png[alt=SmallRye Stork, width=60%, align=center]
+
+To address these issues, we have created https://smallrye.io/smallrye-stork[SmallRye Stork]. Stork provides both features:
+
+- service discovery - it lets you locate the service you need to call;
+- service selection - it lets you select the right one when there is more than one instance
+
+== What is Stork?
+SmallRye Stork is a _service discovery and client-side load-balancing framework_.
+
+Out of the box, it provides integrations with Hashicorp Consul, Eureka, and Kubernetes and brings in a couple of load-balancer (handling the service selection) implementations, such as round-robin. But the most significant strength of Stork is its extensibility. You can plug your own service discovery mechanism or implement your own, maybe business-related, service selection strategy.
+
+== Stork and Quarkus
+As of Quarkus 2.5.0.Final, Quarkus gRPC and the Quarkus Reactive REST Client can use Stork to handle the discovery and the selection of the services.
+
+You just need to configure how you want to locate and select your services and add the related implementation, and the rest will be handled by the clients directly.
```

Comment:
Remove and add the relevant implementations - too many and

Reply:
`You just need to configure how you want to locate and select your services. The rest will be handled by the clients directly.` ?